src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/register/PullRequestStoreImpl.java

```diff
@@ -40,11 +64,29 @@ public boolean hasOpenPullRequests(String branchName, BitbucketSCMRepository rep
                 .isPresent();
     }
 
+    /**
+     * retrieves a pull request given ids and keys
+     * @param key
+     * @param slug
+     * @param serverId
+     * @param requestId
+     * @return desired pull request else null
+     */
     @Override
-    public ConcurrentMap<PullRequestStoreImpl.CacheKey, ConcurrentLinkedQueue<BitbucketPullRequest>> getPullRequests() {
-        return pullRequests;
+    public BitbucketPullRequest getPullRequest(String key, String slug, String serverId, int requestId) {
+        PullRequestStoreImpl.CacheKey cacheKey =
+                new PullRequestStoreImpl.CacheKey(key, slug, serverId);
+        for (BitbucketPullRequest b: pullRequests.get(cacheKey)) {
+            if (b.getId() == requestId) {
+                return b;
+            }
+        }
+        return null;
     }
 
+    /**
+     * key for the store that distinguishes between pull requests within different repos/projects/servers
+     */
     static class CacheKey {
```

Comment:
This class should be `private` the only usages I found was in the test file and they are unused at that, so I think we can make this private now.